rmc/utils/constraint.py

```diff
@@ -288,50 +309,32 @@ def create_filtered_context_ht(
         overwrite=overwrite,
     )
 
-    # TODO: Import models built in gnomad-constraint (update models at resource path)
-    if build_models_from_scratch:
-        logger.info("Building plateau and coverage models...")
-        coverage_ht = prop_obs_coverage.ht()
-        coverage_x_ht = hl.read_table(prop_obs_coverage.path.replace(".ht", "_x.ht"))
-        coverage_y_ht = hl.read_table(prop_obs_coverage.path.replace(".ht", "_y.ht"))
-        (
-            coverage_model,
-            plateau_models,
-            plateau_x_models,
-            plateau_y_models,
-        ) = generate_models(
-            coverage_ht,
-            coverage_x_ht,
-            coverage_y_ht,
-        )
-        # Write out models to HailExpression to save
-        hl.experimental.write_expression(
-            hl.struct(
-                coverage=coverage_model,
-                plateau_models=plateau_models,
-                plateau_X=plateau_x_models,
-                plateau_Y=plateau_y_models,
-            ),
-            coverage_plateau_models_path,
-            overwrite=overwrite,
-        )
-    check_file_exists_raise_error(
-        coverage_plateau_models_path,
-        error_if_not_exists=True,
-        error_if_not_exists_msg=(
-            "Coverage and plateau models HailExpression does not exist!"
-            " Please double check and/or rerun with"
-            " `build_models_from_scratch` = True"
-        ),
+    logger.info(
+        "Counting number of possible variants by variant type + transcript grouping..."
+    )
+    possible_ht = count_variants_by_group(
+        ht,
+        additional_grouping=additional_grouping,
+        use_table_group_by=True,
+    )
+    mu_ht = hl.read_table(
+        get_mutation_ht().path, _n_partitions=mu_ht_partitions
+    ).select("mu_snp")
+    possible_ht = annotate_with_mu(possible_ht, mu_ht)
+    possible_ht = possible_ht.transmute(possible_variants=possible_ht.variant_count)
+    possible_ht = possible_ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/possible_variants.ht",
+        _read_if_exists=not overwrite,
+        overwrite=overwrite,
```

Comment:
I don't see `possible_ht` get used - is this supposed to be joined with `ht` before passing into `calculate_exp_from_mu`? I'm confused on how the `possible_variants` annotation is used

Reply:
ooh. this got lost in my porting from a notebook back into the repo. the join is (conceptually) done in `calculate_exp_from_mu` but I forgot to actually pass the tables properly to do this